### ISSUE TITLE
Error installing Nix in Mac with FileVaut encrypted volumes

### ISSUE THREAD

**Issue by pupoxcu** [digitallyinduced/ihp#93]
Following the install guide executed: `sh <(curl https://nixos.org/nix/install) --darwin-use-unencrypted-nix-store-volume` then it returned: `error: refusing to create Nix store volume because the boot volume is FileVault encrypted, but encryption-at-rest is not available. Manually create a volume for the store and re-run this script. See https://nixos.org/nix/manual/#sect-macos-installation`

**Comment by mpscholten** [comment]
What macbook version are you using?

Can you try these steps (take from https://github.com/NixOS/nix/issues/2925#issuecomment-539490866):

```
sudo mkdir /System/Volumes/Data/opt/nix
sudo chown {your_user} /System/Volumes/Data/opt/nix
sudo sh -c "echo 'nix\tSystem/Volumes/Data/opt/nix' >> /etc/synthetic.conf"
```
Reboot mac. Then:
```
export NIX_IGNORE_SYMLINK_STORE=1
curl https://nixos.org/nix/install | sh     
```

After that, add the following to `~/.zprofile`:

```
. ~/.nix-profile/etc/profile.d/nix.sh
export NIX_IGNORE_SYMLINK_STORE=1
```

I've just tried these steps on a clean 2018 macbook pro (where the above nix installation error message also appeared), and this fixed the problem.

**Comment by ajkovar** [comment]
This didn't work for me as is.  I had to add the -L argument to curl.  I guess the page moved? 🤷‍♂️

   export NIX_IGNORE_SYMLINK_STORE=1
   curl https://nixos.org/nix/install | sh

**Comment by jantebeest** [comment]
Doesn't work on MacOS 11, the check looks to be insufficient as it tells me to _upgrade_ to 10.x

**Comment by mpscholten** [comment]
What commands/approach did you try? And what's the error message?

**Comment by jantebeest** [comment]
```➜  ~ sh <(curl -L https://nixos.org/nix/install) --darwin-use-unencrypted-nix-store-volume
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
100  2490  100  2490    0     0    771      0  0:00:03  0:00:03 --:--:--   771
downloading Nix 2.3.7 binary tarball for x86_64-darwin from 'https://releases.nixos.org/nix/nix-2.3.7/nix-2.3.7-x86_64-darwin.tar.xz' to '/var/folders/ww/szc9b_qj073659xpp7p8mcbc0000gn/T/nix-binary-tarball-unpack.XXXXXXXXXX.ZstpgXxM'...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 26.4M  100 26.4M    0     0  5903k      0  0:00:04  0:00:04 --:--:-- 6373k
/var/folders/ww/szc9b_qj073659xpp7p8mcbc0000gn/T/nix-binary-tarball-unpack.XXXXXXXXXX.ZstpgXxM/unpack/nix-2.3.7-x86_64-darwin/install: macOS 11.0 is not supported, upgrade to 10.12.6 or higher```

**Comment by jantebeest** [comment]
11.0 is not supported, upgrade to 10.12.6 lol

**Comment by jantebeest** [comment]
Sorry, command got stripped somehow ...
```➜  ~ sh <(curl -L https://nixos.org/nix/install) --darwin-use-unencrypted-nix-store-volume```

**Comment by hk7alone** [comment]
N

**Comment by damienxy** [comment]
FYI, I didn't try the above solution, but [this blog post ](https://www.philipp.haussleiter.de/2020/04/fixing-nix-setup-on-macos-catalina/) worked well for me (Macbook Pro 2017 (Pre-T2) Catalina with FileVault).

**Comment by mpscholten** [comment]
This issue has been finally fixed by nix. The macos install script has been updated to automatically resolve the problem here.

See https://github.com/NixOS/nix/pull/4289

**Comment by abathur** [comment]
Just for completeness/confusion-avoidance, it'll still need to be released (either in 2.4, or backported to 2.3.x) before it'll work for standard installs. There are instructions in the PR head comment for using a test installer generated from it, which may be a tolerable stopgap (but no substitute for squeaking loudly about getting a new release cut? :)
